### PR TITLE
Fix dynamic batching

### PR DESCRIPTION
**Pull request by Adel-Moumen**
This PR aims at fixing dynamic batching in SpeechBrain. Indeed, as explained in #1881, some parameters are not actually passed in the `DynamicBatchSampler` object.  

This PR is also a clean up of #1891.

### CODE REVIEW

**Comment by Adel-Moumen** [comment]
What should we do about:
``` 
Furthermore, "length" as a variable name seems inconclusive as there were two args (length as duration & length as number of examples). As a result, the max_batch_len can be confused with max_batch_ex. The former term has been renamed here to make clear, it is about the duration. This change breaks the interface of the dynamic batch sampler, which is relevant only during training, for only few recipes (five of them have dynamic batching on). If that's an issue -> we can put this PR also on the unstable-v0.6 branch.
```  

?

Would you like to change the name? CC: @mravanelli @TParcollet

File: speechbrain/dataio/sampler.py [hunk 1]
```diff
@@ -485,7 +486,13 @@ def __init__(
         self._max_batch_ex = max_batch_ex
         # Calculate bucket lengths - how often does one bucket boundary fit into max_batch_length?
         self._bucket_lens = [
-            max(1, int(max_batch_length / self._bucket_boundaries[i]))
+            min(
```
Comment (TParcollet) [inline speechbrain/dataio/sampler.py:489]: This change is good to me.

File: speechbrain/dataio/sampler.py [hunk 2]
```diff
@@ -578,6 +585,15 @@ def _generate_batches(self):
             item_len = self._ex_lengths[str(idx)]
             # bucket to fill up most padding
             bucket_id = np.searchsorted(self._bucket_boundaries, item_len)
+
+            if (
+                len(bucket_batches[bucket_id])
```
Comment (TParcollet) [inline speechbrain/dataio/sampler.py:590]: This change is a bit more mysterious to me

Comment (Adel-Moumen) [reply speechbrain/dataio/sampler.py:590]: This is this code: https://github.com/speechbrain/speechbrain/blob/develop/speechbrain/dataio/sampler.py#L595  that has been moved here following Andreas PR.

**Pull request merged**